### ISSUE TITLE
How to specify measure range displayed?

### ISSUE THREAD

**Issue by dlujan** [opensheetmusicdisplay/opensheetmusicdisplay#567]
"For the second main feature, you can now display arbitrary bars of a piece. For example, if you only want to see and practice bars 9-15 of a piece, you can now display them on OSDM without any hassle. It is as simple as it sounds. Just put in the bars that you want to display on our site and off it goes!"

I have no idea how this is actually doable. I don't see a place to change this in the demo or the getting started build. I would like to be able to specify measure ranges in my code if that's possible.
Thanks

**Comment by sschmidTU** [comment]
Like this:

```typescript
openSheetMusicDisplay.setOptions({
  drawFromMeasureNumber: 9,
  drawUpToMeasureNumber: 15
});
openSheetMusicDisplay.render();
```

See the demo code, which is in [index.js](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/develop/demo/index.js).

For further guidance for OSMD, look at the [Wiki](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/wiki) (check Getting started) or feel free to ask us here!

**Comment by dlujan** [comment]
Sorry, I'm asking so many questions. But this technology is really exciting!! Where can I find documentation for the other methods I can use within setOptions, for example? I'm currently looking under Class Documentation. There's just so many things listed!

Thanks

**Comment by sschmidTU** [comment]
> Is it possible to not have it create a double bar line at the end of a given range?
It is definitely desirable! It looks like you found unintended behavior, aka a bug.
When OSMD renders a range of measures, the last barline should be the one in the originally, not automatically a double barline. I've opened an issue for it, #569.

For information on the OSMDOptions, I would check the docstrings of the OSMDOptions interface:
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/develop/src/OpenSheetMusicDisplay/OSMDOptions.ts
For the exact details, i would always look at the code, e.g. OpenSheetMusicDisplay.setOptions() and EngravingRules.

Also, [Getting Started](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/wiki/Getting-Started) in the Wiki has a section on Options, though it's probably too hard to find and should have its own page.

Don't worry, i've answered many more questions from other people and I'm glad to help ;)
Also, it's nice to hear that you're enjoying OSMD!

**Comment by sschmidTU** [comment]
By the way, people keep asking about click events for notes, it would be a nice middle-priority feature to add to OSMD itself, but it might take some time.
We already gave a short hint on how it may be implemented [in #504](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/issues/504#issuecomment-477531178).

**Comment by ifredom** [comment]
> Like this:
> 
> ```ts
> osmd.setOptions({
>   drawFromMeasureNumber: 9,
>   drawUpToMeasureNumber: 15
> });
> osmd.render();
> ```
> 
> See the demo code, which is in [index.js](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/develop/demo/index.js).
> 
> For further guidance for OSMD, look at the [Wiki](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/wiki) (check Getting started) or feel free to ask us here!

good job!
